xbmc/utils/ContentUtils.cpp

```diff
@@ -44,3 +44,18 @@ const std::string ContentUtils::GetPreferredArtImage(const CFileItem& item)
   }
   return item.GetArt("thumb");
 }
+
+CFileItem* ContentUtils::GeneratePlayableTrailerItem(const CFileItemPtr& item,
+                                                     const std::string& label)
+{
+  CFileItem* trailerItem = new CFileItem();
+  trailerItem->SetPath(item->GetVideoInfoTag()->m_strTrailer);
+  *trailerItem->GetVideoInfoTag() = *item->GetVideoInfoTag();
+  trailerItem->GetVideoInfoTag()->m_streamDetails.Reset();
+  trailerItem->GetVideoInfoTag()->m_strTitle =
+      StringUtils::Format("{} ({})", item->GetVideoInfoTag()->m_strTitle, label);
+  trailerItem->SetArt(item->GetArt());
+  trailerItem->GetVideoInfoTag()->m_iDbId = -1;
+  trailerItem->GetVideoInfoTag()->m_iFileId = -1;
```

Comment:
instead of calling `trailerItem-GetVideoInfoTag()` multiple times you could do it only once and assign the return value to a variable. 

Why? You never know how expensive a function can get in the future (example: today it's a one line inline function, later it gets changed to a high complex lazy evaluating non inline function. The method signature stays the same, callers will not be adjusted,  performance drops. I have seen cases like this...)

Reply:
yeap nice point